### PR TITLE
ticket-31: Create training inputs

### PR DESCRIPTION
**Pull request by morriscb**

### CODE REVIEW

**Review by aamster — CHANGES_REQUESTED** [review]

File: src/deepcell/cli/modules/create_train_test_split.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            # Count the number of times this region was classified by a
+            # User.
+            n_classifiers = len(label_results)
+            if n_classifiers < data["min_classifiers"]:
+                continue
+            # Load the labeling data if the number of classifiers is
+            # sufficient.
+            roi_data = [json.loads(label_row[-1])
+                        for label_row in label_results]
+            for roi_idx in range(len(roi_data[0])):
```
[the remark this answers is not in the document]
Comment (morriscb) [reply src/deepcell/cli/modules/create_train_test_split.py:92]: Sounds good. Figured there was potentially enforced ordering from the app but if not I can be more careful with these checks.

File: tests/cli/modules/test_create_train_test_split.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        train_test.run()
+
+        with open(Path(self.artifact_dir.name) / "train_rois.json") as jfile:
+            train_rois = json.load(jfile)
+        with open(Path(self.artifact_dir.name) / "test_rois.json") as jfile:
+            test_rois = json.load(jfile)
+
+        for roi_idx, roi in enumerate(train_rois):
+            if roi_idx > 0:
+                roi['label'] == 'cell'
```
[the remark this answers is not in the document]
Comment (morriscb) [reply tests/cli/modules/test_create_train_test_split.py:165]: Sorry, missed putting in the assert.

File: src/deepcell/cli/modules/create_train_test_split.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (morriscb) [reply src/deepcell/cli/modules/create_train_test_split.py]: Finished up changing the logic here. Now I load the ROI data from json into a DataFrame indexed by `roi_id`.

File: src/deepcell/cli/modules/create_train_test_split.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+            region_labels = [pd.DataFrame(
+                                 data=json.loads(labels)).set_index('roi_id')
+                             for _, _, labels in label_results]
+            for roi_id, roi_row in region_labels[0].iterrows():
+                # Don't create a ModelInput if the label is a False Negative.
+                if not roi_row['is_segmented']:
+                    continue
+                # Count the number of votes for "cell". If the percentage
+                # of votes is sufficient, set the ROI label to cell.
+                n_votes = 0
```
Comment (aamster) [inline src/deepcell/cli/modules/create_train_test_split.py:101]: This variable not needed anymore

File: src/deepcell/cli/modules/create_train_test_split.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+            # Load the labeling data if the number of classifiers is
+            # sufficient.
+            region_labels = [pd.DataFrame(
+                                 data=json.loads(labels)).set_index('roi_id')
+                             for _, _, labels in label_results]
+            for roi_id, roi_row in region_labels[0].iterrows():
+                # Don't create a ModelInput if the label is a False Negative.
+                if not roi_row['is_segmented']:
+                    continue
+                # Count the number of votes for "cell". If the percentage
```
Comment (aamster) [inline src/deepcell/cli/modules/create_train_test_split.py:99]: Please remove this doc; it's not needed anymore since this has been modularized

File: src/deepcell/cli/modules/create_train_test_split.py [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+        data['model_inputs'] = model_inputs
+        return data
+
+    @staticmethod
+    def _tally_votes(roi_id: int,
+                     region_labels: List[pd.DataFrame],
+                     n_labelers: int,
+                     min_labelers_per_region: int,
+                     vote_threshold: int):
+        """Loop through labels for a given roi_id and return if they were
```
Comment (aamster) [inline src/deepcell/cli/modules/create_train_test_split.py:126]: That's not actually what this method is doing; it is returning a final label given multiple labeler's votes for a given roi

**Review by aamster — APPROVED** [review]
Please address the remaining comments. Otherwise looks good.

**Comment by morriscb** [comment]
Hey, had to fix up how the experiment metadata gets ingested. Check out the changes in `get_experiment_metadata`. That's the simplest solution I came up with without adding a significant number of unused and optional parameters to the experiment_metadata schema. Let me know if you're happy with this or if I should add all the unused columns to the schema.

File: src/deepcell/datasets/model_input.py [hunk 6]
```diff
@@ -84,7 +84,7 @@ def project_name(self) -> str:
 
     @classmethod
     def from_data_dir(cls, data_dir: Union[str, Path], experiment_id: str,
-                      roi_id: str):
+                      roi_id: str, label: Optional[str] = None):
         """Instantiate a ModelInput from a data_dir
```
[the remark this answers is not in the document]
Comment (aamster) [reply src/deepcell/datasets/model_input.py:88]: @morriscb please also remove the "This should be used at test time", since you are using it to build training examples with labels

**Comment by aamster** [comment]
@morriscb that solution looks fine
